### PR TITLE
Allow multi-step settings in custodian

### PR DESCRIPTION
**Pull request by NicolasDorier**
Ping @woutersamaey

### CODE REVIEW

File: BTCPayServer/Models/CustodianAccountViewModels/EditCustodianAccountViewModel.cs [hunk 1]
```diff
@@ -8,5 +8,6 @@ public class EditCustodianAccountViewModel
 
         public CustodianAccountData CustodianAccount { get; set; }
         public Form ConfigForm { get; set; }
+        public string Blob  { get; set; }
```
Comment (woutersamaey) [inline BTCPayServer/Models/CustodianAccountViewModels/EditCustodianAccountViewModel.cs:11]: What does this Blob do? It's just a hidden field on the form?

File: BTCPayServer.Abstractions/Custodians/ICustodian.cs [hunk 2]
```diff
@@ -20,6 +21,6 @@ public interface ICustodian
      */
     Task<Dictionary<string, decimal>> GetAssetBalancesAsync(JObject config, CancellationToken cancellationToken);
 
-    public Task<Form.Form> GetConfigForm(CancellationToken cancellationToken = default);
+    public Task<Form.Form> GetConfigForm(JObject config, CancellationToken cancellationToken = default);
```
Comment (woutersamaey) [inline BTCPayServer.Abstractions/Custodians/ICustodian.cs:24]: So happy this is back. There's just no other way to go about this.

File: BTCPayServer/Controllers/UICustodianAccountsController.cs [hunk 3]
```diff
@@ -46,7 +47,8 @@ public class UICustodianAccountsController : Controller
             IEnumerable<ICustodian> custodianRegistry,
             BTCPayServerClient btcPayServerClient,
             BTCPayNetworkProvider networkProvider,
-            LinkGenerator linkGenerator
+            LinkGenerator linkGenerator,
+            FormDataService formDataService
```
Comment (woutersamaey) [inline BTCPayServer/Controllers/UICustodianAccountsController.cs:51]: What does this service do? Looks like it's never called, only injected.

Comment (NicolasDorier) [reply BTCPayServer/Controllers/UICustodianAccountsController.cs:51]: It is called, it is used to validate fields in `NextForm`

File: BTCPayServer/Models/CustodianAccountViewModels/EditCustodianAccountViewModel.cs (hunk 1 above)
Comment (NicolasDorier) [reply BTCPayServer/Models/CustodianAccountViewModels/EditCustodianAccountViewModel.cs:11]: It remembers the state of the config between the different steps.

**Comment by NicolasDorier** [comment]
@woutersamaey please can you try again I should have fixed your points.

**Comment by woutersamaey** [comment]
@NicolasDorier It is working now, but you forgot to add the 1 line I asked which fixes the inability to change the custodian's name (the 1 field outside the JObject and outside the form)

> To fix, simply add `custodianAccount.Name = vm.CustodianAccount.Name;` in 
 BTCPayServer/Controllers/UICustodianAccountsController.cs on line 249

**Comment by woutersamaey** [comment]
Good to merge

**Pull request merged**